svgdigitizer/electrochemistry/cv.py

```diff
@@ -814,28 +981,39 @@ def metadata(self):
             ...   </g>
             ...   <text x="-200" y="330">scan rate: 50 V/s</text>
             ...   <text x="-400" y="430">comment: noisy data</text>
-            ...   <text x="-400" y="430">linked: SXRD, SHG</text>
+            ...   <text x="-400" y="530">linked: SXRD, SHG</text>
+            ...   <text x="-200" y="630">Figure: 2b</text>
+            ...   <text x="-200" y="730">tags: BCV, HER, OER</text>
             ... </svg>'''))
             >>> cv = CV(SVGPlot(svg))
             >>> cv.metadata  # doctest: +NORMALIZE_WHITESPACE
-            {'figure description': {'version': 1,
-              'type': 'digitized',
-              'simultaneous measurements': ['SXRD', 'SHG'],
-              'measurement type': 'CV',
-              'scan rate': {'value': 50.0, 'unit': 'V / s'},
-              'axes': {'E': {'unit': 'mV', 'reference': 'RHE', 'orientation': 'x'},
-               'j': {'unit': 'uA / cm2', 'orientation': 'y'},
-               't': {'unit': 's'}},
-              'comment': 'noisy data'},
-             'data description': {'version': 1,
-              'type': 'digitized',
-              'measurement type': 'CV',
-              'axes': {'E': {'unit': 'V', 'reference': 'RHE'},
-               'j': {'unit': 'A / m2'},
-               't': {'unit': 's'}}}}
+            {'experimental': {'tags': ['BCV', 'HER', 'OER']},
+             'source': {'figure': '2b', 'curve': '0'},
+             'figure description': {'version': 1,
+             'type': 'digitized',
+             'simultaneous measurements': ['SXRD', 'SHG'],
+             'measurement type': 'CV',
+             'scan rate': {'value': 50.0, 'unit': 'V / s'},
+             'fields': [{'name': 'E', 'unit': 'mV',
+                        'reference': 'RHE', 'orientation': 'x'},
+                        {'name': 'j', 'unit': 'uA / cm2',
+                        'orientation': 'y'}],
+                        'comment': 'noisy data'},
+             'data description': {'version': 1, 'type': 'digitized',
+                                  'measurement type': 'CV',
+                                  'fields': [{'name': 't', 'unit': 's'},
+                                  {'name': 'E', 'unit': 'V', 'reference': 'RHE'},
+                                  {'name': 'j', 'unit': 'A / m2'}]}}
 
         """
         metadata = self._metadata.copy()
+        # Add experimental to metadata
+        metadata.setdefault("experimental", {})
+        metadata["experimental"]["tags"] = self.tags
+        # Add source to metadata
+        metadata.setdefault("source", {})
+        metadata["source"]["figure"] = self.figure_label
```

Comment:
These will override values from the yaml even if `self.*_label==""`.

Reply:
see above

Reply:
I added an exception since this is actually mandatory.

Reply:
This will cause some trouble. Actually, it is only mandatory in echemdb. So we should check it there.